src/cache/cache_las.c

```diff
@@ -787,6 +788,11 @@ __wt_las_insert_block(WT_CURSOR *cursor,
 		__las_insert_block_verbose(session, btree, multi);
 	}
 
+	las_btree = S2BT_SAFE(session);
+	if (las_btree != NULL)
+		WT_STAT_CONN_SET(session, lookaside_bytes,
+		    las_btree->bm->block->size);
```

Comment:
There is a parameter called `btree` in this function. To my understanding, this BTree is associated with the table owning the page that is being evicted and written to the lookaside file rather than the actual lookaside file itself. So I think setting the statistic to `btree->bm->block->size` would be wrong.

Instead, I've taken the BTree associated with `connection` which comes from the lookaside cursor which seems like it should give me the right value.

Reply:
8150be1

Ok done. I've also wrapped the `__wt_block_manager_named_size` call with `WT_RET`. Not 100% sure whether it warrants returning non-zero since it's just for incrementing a statistic. But I imagine if this doesn't work then something has gone horribly wrong.

If it's easy to explain, what are the layering violations you're talking about? I see you mentioned something similar in a comment on the technical design doc for "mitigate operational issues with pinned history". I had a talk to Vamsi about this before writing this code and we thought that it'd be preferable to get this info from the block manager if it already exists instead of going to the operating system for it.

Reply:
The block manager is the layer between the access methods and the OS.

It should be possible to replace block managers — for example with an NVM friendly engine.

Reply:
@keithbostic 
Ah, understood. So "layering violation" refers to relying on some detail that should be abstracted away at that layer.